Some code improvements

BlackCorsair opened the pull request:
Hi,
I saw your project from reddit and I really liked it!

I'm doing this pull request with some changes in your code (without changing the functionality) just to improve readability and code-quality.

Please don't take this as a critic to your code (it's *your* code and you do with it what you want), I did it with my best intentions to improve it.

Best regards

BlackCorsair committed:
Refactor following PEP8

BlackCorsair committed:
Change isInt for native isinstance

BlackCorsair committed:
Change class comparisson with native isinstance

BlackCorsair committed:
Add requirements.txt

BlackCorsair committed:
Change sys.exec_info for native Exception

BlackCorsair committed:
Change full name of Named, Autocomplete for a short one

BlackCorsair committed:
Create function for repeated code 'downloadCard'

BlackCorsair committed:
Revert "Remove binaries from repo"

This reverts commit 9eb6ae6428b8a283f40208026906bf1759971842.

BlackCorsair committed:
Remove binaries from repo

BlackCorsair committed:
Refactor RobustSearch

BlackCorsair committed:
Change minor smellcodes

BlackCorsair committed:
Format main making it more readable

BlackCorsair committed:
Refactor main making it more readable

TallPatrick commented on imagePicker.py:
```diff
@@ -113,9 +115,9 @@ def main():
-    if str(type(AutoOrCard)) == "<class 'scrython.cards.named.Named'>":
+    if isinstance(AutoOrCard, Named):
         return AutoOrCard.name()
-    if str(type(AutoOrCard)) == "<class 'scrython.cards.autocomplete.Autocomplete'>":
```
OK, this makes sense now, and is also why I was having trouble getting this to work.  I didn't have the correct imports.  I knew my workaround was hacky AF, so this looks a lot more like what I expected it to look like.  Thank you!

TallPatrick commented on imagePicker.py:
```diff
@@ -29,12 +29,12 @@ def main():
         if Card_Query.lower() == 'clear' or Card_Query.lower() == '':
             # clear the image and loop
             print('Resetting image...')
-            shutil.copy2('blank.png', 'card.png')
```
So the reason that that there are these placeholder blank cards is that the way OBS works, you need to point it at something for it to watch for updates.  If the file ever disappears or is deleted, OBS seems to complain and not check for updates and display the picture.  That's why the blank.png is a 1x1 transparent image, and instead of deleting the card.png, it is replaced with the 1x1 transparent one.

This also would allow for someone (if they wanted to) to show some other image when not searching up a card by just replacing the blank.png

So yes, I agree that it's cleaner the way you suggested, but the way that OBS works seems to necessitate that there always be an image there, and copying from a reference 1x1 transparent png is a lot easier than image manipulation by code.

BlackCorsair replied:
Aaaah... so that's why you didn't nuke the file. I clearly didn't read the full README (specially the ADDITIONAL INFO). 
Sorry for that rogue change in the functionality, as my changes didn't intend to affect it.

TallPatrick commented on imagePicker.py:
```diff
@@ -120,44 +120,47 @@ def GetName(AutoOrCard):
+        print("Something went wrong. Returning to prompt.")
+        print(f"error details: {e}")
+        print("\n\n")
+        remove('card.png')
```
Per previous comment, this would have to revert back as well.

TallPatrick commented on imagePicker.py:
```diff
@@ -120,44 +120,47 @@ def GetName(AutoOrCard):
+
+    while retries != 0 or not auto:
         time.sleep(0.05)
         auto = Autocomplete(q=CardQuery.lower(), query=CardQuery.lower())
```
I will have to dig in more to Scryfall's API to understand better what could cause autocomplete search failures.  Unless it is a network issue, failing to search by autocomplete once will fail all 3 times, so this will likely only slow things down in the case of not identifying a card on the first try.

BlackCorsair replied:
I also thought about doing the retries *if* there was a TimeOut exception, but didn't find any specific exceptions in the API and didn't have the time to force a timeout during the initialization.

In the API Github page, there's a [https://github.com/NandaScott/Scrython/blob/1484f9f81811a8e9fffa6b21de1ee203a962cc2f/scrython/foundation.py#L5](ScryfallError) exception that you may use, but I'm not entirely sure if it raises only when there's a network issue.

TallPatrick replied:
Yeah, I looked at that example, and the only exceptions I ever got was errors for not finding cards from a search.  I could dig in to the scryfall API documentation to see if they have a time-out response.  Because if they don't then the time-out exception would be a non-scrython exception and then you could decide to re-try based on that maybe?

Anyways, more research is needed.

TallPatrick reviewed (changes requested):
Overall all this feedback is awesome!  I agree that it looks a lot cleaner with these changes.  The only thing that I think needs to stay is the way the card image is set back to a copy of blank.png, but that's just because of how OBS works.  I'll pull down this branch tomorrow and see if OBS will work with the card image being removed and re-added, but I think it hasn't worked that way in the past.

Additionally, the retry code, while probably correct in most cases is probably not necessary for the auto-complete search.

Again though, thank you for trudging through my awful code and providing suggestions to make it much less awful.

TallPatrick committed:
Revert "Refactor RobustSearch"

This reverts commit 656a7c73d5bb85e6a7156ea50b33fb97709ef3df.

TallPatrick committed:
v1.1 Patch upgrade

Fixing the issues found and suggested by BlackCorsair while also reverting a few of those changes that would break intended functionality

TallPatrick committed:
fixing a typo

Fixing a type
